ql/src/java/org/apache/hadoop/hive/ql/stats/StatsUpdaterThread.java

```diff
@@ -214,23 +219,28 @@ private void stopWorkers() {
       throws MetaException, NoSuchTxnException, NoSuchObjectException {
     if (isAnalyzeTableInProgress(fullTableName)) return null;
     String cat = fullTableName.getCat(), db = fullTableName.getDb(), tbl = fullTableName.getTable();
+    String dbName = MetaStoreUtils.prependCatalogToDbName(cat,db, conf);
+    if (!isDbTargetOfReplication.containsKey(dbName) || !isDbBeingFailedOver.containsKey(dbName)) {
+      Database database = rs.getDatabase(cat, db);
+      isDbTargetOfReplication.put(dbName, ReplUtils.isTargetOfReplication(database));
+      isDbBeingFailedOver.put(dbName, MetaStoreUtils.isDbBeingFailedOver(database));
```

Comment:
Why do we need two separate maps, we don;t need the reason for skip, just tracking what to skip is fine no?